singertools/infer_schema.py

```diff
@@ -90,10 +92,46 @@ def to_json_schema(obs):
     return result
 
 
-def main():
-    for line in sys.stdin:
+def infer_schemas(record_inputs, out_dir):
+    """
+    The main logic that iterates record_inputs and prints the resulting
+    inferred schema to either outdir or stdout
+    """
+    streams = {}
+
+    for line in record_inputs:
         rec = json.loads(line)
         if rec['type'] == 'RECORD':
-            add_observations([], rec['record'])
+            stream = rec['stream']
+            if stream not in streams:
+                streams[stream] = {}
+            streams[stream] = add_observations(streams[stream], [], rec['record'])
```

Comment:
This could look like

``` python
streams = defaultdict(dict)
...
stream = rec['stream']
streams[stream] = add_observations(streams[stream], [], rec['record'])
```